src/views/login/Login.tsx

```diff
@@ -17,13 +18,23 @@ import { loginUserSchema } from "../../utils/validation/schema";
 import type { Login } from "../../types/auth.types";
 import type { ClientSafeProvider } from "next-auth/react";
 
-interface LoginProps {
-  readonly providers: Record<AUTH_PROVIDER, ClientSafeProvider>;
-}
-
-export const LoginView = memo<LoginProps>(({ providers }) => {
+export const LoginView = memo(({}) => {
   const router = useRouter();
+
   const [isFormValidated, setIsFormValidated] = useState(false);
+
+  const [providers, setProviders] = useState<Record<
+    AUTH_PROVIDER,
+    ClientSafeProvider
+  > | null>(null);
+
+  useEffect(() => {
+    void (async () => {
+      const res = await getProviders();
+      setProviders(res);
+    })();
+  }, []);
```

Comment:
Why do we need to move this? I mean we can do simply `const providers = await getProviders();` and it should work in Server Component also